### DISCUSSION TITLE
Call a handler always when the operator starts

### DISCUSSION THREAD

**Discussion by mirza-ali-ctct** [nolar/kopf#985] [General]
Is there any handler which is always called when the operator starts?

**Comment by nolar** [comment]
`@kopf.on.startup`?

**Comment by mirza-ali-ctct** [comment]
The `@kopf.on.startup` is for the whole operator. Is there a way to have startup logic for each resource? Our uses case is to always sync when the pod is restarted. I am using `@kopf.on.resume` and it is executed after a certain delay. I guess, is `@kopf.on.resume`  the recommended way in this case.

**Reply by nolar** [reply]
Yes, `@kopf.on.resume` is the solution for this case. It should run as soon as the resource becomes visible to the operator, usually with zero delay.
